### PR TITLE
docs: added details signifying that node fields can be arbitrary

### PR DESCRIPTION
**Pull request by ElijahCano33**
## Description
The changes introduced by this PR include some changes that were asked to be made 
to the https://www.gatsbyjs.org/docs/data-fetching/ page. One change was to add a comment
in the code example stating that the nameWithOwner and url fields are arbitrary fields from the data, and another change was to add a line explicitly stating that node fields can be arbitrary.

### Documentation
This issue is concerned with the Gatsby GraphQL docs and is located on this [page](https://www.gatsbyjs.org/docs/data-fetching/)

## Related Issues
Addresses #21962

### CODE REVIEW

File: docs/docs/data-fetching.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ exports.sourceNodes = async ({
   const resultData = await result.json()
   // create node for build time data example in the docs
   createNode({
+    //nameWithOwner and url are arbitrary fields from the data
```
Comment (gillkyle) [inline docs/docs/data-fetching.md:77]:
```suggestion
    // nameWithOwner and url are arbitrary fields from the data
```

This is nit-picky but I like seeing these comments as consistent. I'll see if I can push to your branch to make an update like this 🙂

File: docs/docs/data-fetching.md [hunk 2]
```diff
@@ -88,8 +89,11 @@ exports.sourceNodes = async ({
 }
 ```
 
+It is important to note that node fields can indeed be arbitrary. The nameWithOwner and url fields from above are examples of this.
```
Comment (gillkyle) [inline docs/docs/data-fetching.md:92]:
```suggestion
It is important to note that node fields can indeed be arbitrary. The `nameWithOwner` and `url` fields from above are examples of this.
```

Another nit pick, I like seeing code references wrapped in backticks when possible so they render as `url` and `nameWithOwner`, this another thing I can update!

**Comment by gillkyle** [comment]
Alright I don't think I have the ability to push to your fork of the repo 🤔 

Could you make the updates with the suggestions I added? (you can click on the Files changed tab and then add the changes to be committed straight from the GitHub UI) _or_ try and enable this option?

image

**Comment by ElijahCano33** [comment]
Hi gillkyle, thank you for the suggestions. I have committed them just as you have asked.

File: docs/docs/data-fetching.md [hunk 3]
```diff
@@ -88,8 +89,11 @@ exports.sourceNodes = async ({
 }
 ```
 
+It is important to note that node fields can indeed be arbitrary. The `nameWithOwner` and `url` fields from above are examples of this.
+
 This node created manually could be retrieved with a query like this:
 
+
```
Comment (gillkyle) [inline docs/docs/data-fetching.md:96]:
```suggestion
```

It looks like this line is likely what's causing the linting tests to fail, could you remove the empty new line as well?

**Comment by ElijahCano33** [comment]
Thank you once again, and it appears that the lint test has indeed passed successfully now after the change. Cheers!

**Pull request merged**